internal/services/azuremanagedlustrefilesystem/managed_lustre_file_system_resource.go

```diff
@@ -0,0 +1,484 @@
+
+func (r ManagedLustreFileSystemResource) ResourceType() string {
+	return "azurerm_managed_lustre_file_system"
+}
+
+func (r ManagedLustreFileSystemResource) ModelObject() interface{} {
+	return &ManagedLustreFileSystemModel{}
+}
+
+func (r ManagedLustreFileSystemResource) IDValidationFunc() pluginsdk.SchemaValidateFunc {
+	return amlfilesystems.ValidateAmlFilesystemID
+}
+
+func (r ManagedLustreFileSystemResource) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"name": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validate.ManagedLustreFileSystemName,
+		},
+
+		"resource_group_name": commonschema.ResourceGroupName(),
+
+		"location": commonschema.Location(),
+
+		"maintenance_window": {
+			Type:     pluginsdk.TypeList,
+			Required: true,
+			MaxItems: 1,
+			Elem: &pluginsdk.Resource{
+				Schema: map[string]*pluginsdk.Schema{
+					"day_of_week": {
+						Type:         pluginsdk.TypeString,
+						Required:     true,
+						ValidateFunc: validation.StringInSlice(amlfilesystems.PossibleValuesForMaintenanceDayOfWeekType(), false),
+					},
+
+					"time_of_day_in_utc": {
+						Type:         pluginsdk.TypeString,
+						Required:     true,
+						ValidateFunc: validate.TimeOfDayInUTC,
+					},
+				},
+			},
+		},
+
+		"sku_name": {
+			Type:     pluginsdk.TypeString,
+			Required: true,
+			ForceNew: true,
+			ValidateFunc: validation.StringInSlice([]string{
+				"AMLFS-Durable-Premium-40",
+				"AMLFS-Durable-Premium-125",
+				"AMLFS-Durable-Premium-250",
+				"AMLFS-Durable-Premium-500",
+			}, false),
+		},
+
+		"storage_capacity_in_tb": {
+			Type:     pluginsdk.TypeInt,
+			Required: true,
+			ForceNew: true,
+			ValidateFunc: validation.All(
+				validation.IntBetween(8, 128),
+				validation.IntDivisibleBy(8),
+			),
+		},
+
+		"subnet_id": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: commonids.ValidateSubnetID,
+		},
+
+		"zones": commonschema.ZonesMultipleRequiredForceNew(),
+
+		"hsm_setting": {
+			Type:     pluginsdk.TypeList,
+			Optional: true,
+			ForceNew: true,
+			MaxItems: 1,
+			Elem: &pluginsdk.Resource{
+				Schema: map[string]*pluginsdk.Schema{
+					"container_id": {
+						Type:         pluginsdk.TypeString,
+						Required:     true,
+						ForceNew:     true,
+						ValidateFunc: storageValidate.StorageContainerResourceManagerID,
+					},
+
+					"logging_container_id": {
+						Type:         pluginsdk.TypeString,
+						Required:     true,
+						ForceNew:     true,
+						ValidateFunc: storageValidate.StorageContainerResourceManagerID,
+					},
+
+					"import_prefix": {
+						Type:         pluginsdk.TypeString,
+						Optional:     true,
+						ForceNew:     true,
+						ValidateFunc: validate.ImportPrefix,
+					},
+				},
+			},
+		},
+
+		"identity": commonschema.UserAssignedIdentityOptionalForceNew(),
+
+		"key_encryption_key": {
```

Comment:
Please rename this to 
```suggestion
		"encryption_key": {
```